File: little_artist_flutter/lib/views/paywall_view.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:in_app_purchase/in_app_purchase.dart';

import '../providers/premium_provider.dart';
import '../services/haptic_service.dart';
import '../utils/brand_tokens.dart';

/// Full-screen paywall that displays a feature comparison between the free and
/// premium tiers and lets the user subscribe via monthly or yearly plans.
class PaywallView extends ConsumerStatefulWidget {
  const PaywallView({super.key});

  @override
  ConsumerState<PaywallView> createState() => _PaywallViewState();
}

class _PaywallViewState extends ConsumerState<PaywallView> {
  ProductDetails? _selectedProduct;

  @override
  Widget build(BuildContext context) {
    final store = ref.watch(storeManagerProvider);
    final monthly = store.monthlyProduct;
    final yearly = store.yearlyProduct;

    // Default to yearly if available.
    _selectedProduct ??= yearly ?? monthly;

    return Scaffold(
      backgroundColor: Brand.cream,
      appBar: AppBar(
        backgroundColor: Colors.transparent,
        elevation: 0,
        leading: IconButton(
          icon: const Icon(Icons.close, color: Brand.charcoal),
          onPressed: () => Navigator.of(context).pop(),
        ),
      ),
      body: SafeArea(
        child: Column(
          children: [
            Expanded(
              child: ListView(
                padding: const EdgeInsets.symmetric(
                  horizontal: Brand.screenPadding,
                ),
                children: [
                  // -------------------------------------------------------------
                  // MARK: - Hero Header
                  // -------------------------------------------------------------
                  _buildHeroHeader(),
                  const SizedBox(height: Brand.sectionSpacing),

                  // -------------------------------------------------------------
                  // MARK: - Feature Comparison
                  // -------------------------------------------------------------
                  _buildFeatureComparison(context),
                  const SizedBox(height: Brand.sectionSpacing),

                  // -------------------------------------------------------------
                  // MARK: - Subscription Cards
                  // -------------------------------------------------------------
                  if (monthly != null)
                    _buildPlanCard(
                      context,
                      product: monthly,
                      label: 'Monthly',
                      isHighlighted: false,
                    ),
                  if (monthly != null && yearly != null)
                    const SizedBox(height: 12),
                  if (yearly != null)
                    _buildPlanCard(
                      context,
                      product: yearly,
                      label: 'Yearly',
                      isHighlighted: true,
                      badge: 'Best Value',
                    ),
                  if (store.products.isEmpty)
                    Padding(
                      padding: const EdgeInsets.symmetric(vertical: 24),
                      child: Text(
                        'Loading plans...',
                        textAlign: TextAlign.center,
                        style: Brand.bodyFont.copyWith(color: Brand.warmGray),
                      ),
                    ),

                  if (store.errorMessage != null) ...[
                    const SizedBox(height: 12),
                    Text(
                      store.errorMessage!,
                      textAlign: TextAlign.center,
                      style: Brand.captionFont.copyWith(color: Brand.dustyRose),
                    ),
                  ],
                  const SizedBox(height: Brand.sectionSpacing),
                ],
              ),
            ),

            // -----------------------------------------------------------------
            // MARK: - Bottom Actions
            // -----------------------------------------------------------------
            _buildBottomActions(context, store),
          ],
        ),
      ),
    );
  }

  // ---------------------------------------------------------------------------
  // MARK: - Hero Header
  // ---------------------------------------------------------------------------

  Widget _buildHeroHeader() {
    return Container(
      width: double.infinity,
      padding: const EdgeInsets.symmetric(vertical: 32),
      decoration: BoxDecoration(
        gradient: const LinearGradient(
          begin: Alignment.topLeft,
          end: Alignment.bottomRight,
          colors: [Brand.primary, Color(0xFFE85D2F)],
        ),
        borderRadius: BorderRadius.circular(Brand.radiusCard),
        boxShadow: Brand.cardShadow,
      ),
      child: Column(
        children: [
          Container(
            width: 64,
            height: 64,
            decoration: BoxDecoration(
              color: Colors.white.withValues(alpha: 0.2),
              shape: BoxShape.circle,
            ),
            child: const Icon(
              Icons.workspace_premium,
              size: 36,
              color: Colors.white,
            ),
          ),
          const SizedBox(height: 16),
          Text(
            'Unlock Premium',
            style: Brand.title1Font.copyWith(color: Colors.white),
          ),
          const SizedBox(height: 8),
          Text(
            'Archive every masterpiece, without limits.',
            style: Brand.bodyFont.copyWith(
              color: Colors.white.withValues(alpha: 0.9),
            ),
            textAlign: TextAlign.center,
          ),
        ],
      ),
    );
  }

  // ---------------------------------------------------------------------------
  // MARK: - Feature Comparison
  // ---------------------------------------------------------------------------

  Widget _buildFeatureComparison(BuildContext context) {
    const features = [
      _FeatureRow('Children', '2', 'Unlimited'),
      _FeatureRow('Artworks', '50', 'Unlimited'),
      _FeatureRow('Family sharing', null, 'Included'),
      _FeatureRow('Voice memos', null, 'Included'),
      _FeatureRow('Cloud sync', null, 'Included'),
    ];

    return Container(
      decoration: BoxDecoration(
        color: Brand.surface,
        borderRadius: BorderRadius.circular(Brand.radiusCard),
        boxShadow: Brand.cardShadow,
      ),
      clipBehavior: Clip.antiAlias,
      child: Column(
        children: [
          // Table header
          Padding(
            padding: const EdgeInsets.fromLTRB(16, 16, 16, 12),
            child: Row(
              children: [
                Expanded(
                  flex: 3,
                  child: Text(
                    'Feature',
                    style: Brand.captionFont.copyWith(
                      color: Brand.warmGray,
                      fontWeight: FontWeight.w600,
                    ),
                  ),
                ),
                Expanded(
                  flex: 2,
                  child: Text(
                    'Free',
                    textAlign: TextAlign.center,
                    style: Brand.captionFont.copyWith(
                      color: Brand.warmGray,
                      fontWeight: FontWeight.w600,
                    ),
                  ),
                ),
                Expanded(
                  flex: 2,
                  child: Text(
                    'Premium',
                    textAlign: TextAlign.center,
                    style: Brand.captionFont.copyWith(
                      color: Brand.primary,
                      fontWeight: FontWeight.w600,
                    ),
                  ),
                ),
              ],
            ),
          ),
          const Divider(height: 1, color: Brand.softTan),
          ...features.map((f) => _buildFeatureRow(context, f)),
        ],
      ),
    );
  }

  Widget _buildFeatureRow(BuildContext context, _FeatureRow feature) {
    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 14),
      child: Row(
        children: [
          Expanded(
            flex: 3,
            child: Text(
              feature.name,
              style: Brand.bodyFont.copyWith(color: Brand.charcoal),
            ),
          ),
          Expanded(
            flex: 2,
            child: feature.freeValue != null
                ? Text(
                    feature.freeValue!,
                    textAlign: TextAlign.center,
                    style: Brand.bodyFont.copyWith(color: Brand.warmGray),
                  )
                : const Icon(Icons.close, size: 20, color: Brand.warmGray),
          ),
          Expanded(
            flex: 2,
            child: feature.premiumValue == 'Included'
                ? const Icon(Icons.check_circle, size: 20, color: Brand.primary)
                : Text(
                    feature.premiumValue,
                    textAlign: TextAlign.center,
                    style: Brand.bodyFont.copyWith(
                      color: Brand.primary,
                      fontWeight: FontWeight.w600,
                    ),
                  ),
          ),
        ],
      ),
    );
  }

  // ---------------------------------------------------------------------------
  // MARK: - Plan Card
  // ---------------------------------------------------------------------------

  Widget _buildPlanCard(
    BuildContext context, {
    required ProductDetails product,
    required String label,
    required bool isHighlighted,
    String? badge,
  }) {
    final isSelected = _selectedProduct?.id == product.id;

    return GestureDetector(
      onTap: () {
        HapticService.light();
        setState(() => _selectedProduct = product);
      },
      child: AnimatedContainer(
        duration: const Duration(milliseconds: 200),
        padding: const EdgeInsets.all(16),
        decoration: BoxDecoration(
          color: isHighlighted ? Brand.primary : Brand.surface,
          borderRadius: BorderRadius.circular(Brand.radiusCard),
          border: Border.all(
            color: isSelected
                ? (isHighlighted ? Colors.white : Brand.primary)
                : Colors.transparent,
            width: 2,
          ),
          boxShadow: Brand.cardShadow,
        ),
        child: Row(
          children: [
            // Radio indicator
            Container(
              width: 24,
              height: 24,
              decoration: BoxDecoration(
                shape: BoxShape.circle,
                border: Border.all(
                  color: isHighlighted
                      ? Colors.white
                      : (isSelected ? Brand.primary : Brand.softTan),
                  width: 2,
                ),
              ),
              child: isSelected
                  ? Center(
                      child: Container(
                        width: 12,
                        height: 12,
                        decoration: BoxDecoration(
                          shape: BoxShape.circle,
                          color: isHighlighted ? Colors.white : Brand.primary,
                        ),
                      ),
                    )
                  : null,
            ),
            const SizedBox(width: 12),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Row(
                    children: [
                      Text(
                        label,
                        style: Brand.headlineFont.copyWith(
                          color:
                              isHighlighted ? Colors.white : Brand.charcoal,
                        ),
                      ),
                      if (badge != null) ...[
                        const SizedBox(width: 8),
                        Container(
                          padding: const EdgeInsets.symmetric(
                            horizontal: 8,
                            vertical: 2,
                          ),
                          decoration: BoxDecoration(
                            color: isHighlighted
                                ? Colors.white.withValues(alpha: 0.25)
                                : Brand.primaryTint,
                            borderRadius: BorderRadius.circular(100),
                          ),
                          child: Text(
                            badge,
                            style: Brand.caption2Font.copyWith(
                              color: isHighlighted
                                  ? Colors.white
                                  : Brand.primary,
                              fontWeight: FontWeight.w600,
                            ),
                          ),
                        ),
                      ],
                    ],
                  ),
                  const SizedBox(height: 2),
                  Text(
                    product.price,
                    style: Brand.bodyFont.copyWith(
                      color: isHighlighted
                          ? Colors.white.withValues(alpha: 0.9)
                          : Brand.warmGray,
                    ),
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }

  // ---------------------------------------------------------------------------
  // MARK: - Bottom Actions
  // ---------------------------------------------------------------------------

  Widget _buildBottomActions(BuildContext context, dynamic store) {
    return Container(
      padding: EdgeInsets.fromLTRB(
        Brand.screenPadding,
        12,
        Brand.screenPadding,
        MediaQuery.of(context).padding.bottom + 12,
      ),
      decoration: BoxDecoration(
        color: Brand.cream,
        border: const Border(
          top: BorderSide(color: Brand.softTan, width: 0.5),
        ),
      ),
      child: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          // Subscribe button
          SizedBox(
            width: double.infinity,
            height: 52,
            child: ElevatedButton(
              onPressed: store.isPurchasing || _selectedProduct == null
                  ? null
                  : () {
                      HapticService.medium();
                      store.purchase(_selectedProduct!);
                    },
              style: ElevatedButton.styleFrom(
                backgroundColor: Brand.primary,
                foregroundColor: Colors.white,
                disabledBackgroundColor: Brand.disabled,
                disabledForegroundColor: Colors.white,
                shape: RoundedRectangleBorder(
                  borderRadius: BorderRadius.circular(Brand.radiusButton),
                ),
                elevation: 0,
              ),
              child: store.isPurchasing
                  ? const SizedBox(
                      width: 22,
                      height: 22,
                      child: CircularProgressIndicator(
                        strokeWidth: 2.5,
                        color: Colors.white,
                      ),
                    )
                  : Text('Subscribe', style: Brand.headlineFont),
            ),
          ),
          const SizedBox(height: 12),

          // Restore purchases
          TextButton(
            onPressed: () {
              HapticService.light();
              ref.read(storeManagerProvider).restorePurchases();
            },
            child: Text(
              'Restore Purchases',
              style: Brand.subheadlineFont.copyWith(color: Brand.warmGray),
            ),
          ),
          const SizedBox(height: 4),

          // Terms & privacy
          Row(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              GestureDetector(
                onTap: () {
                  // Open terms of use
                },
                child: Text(
                  'Terms of Use',
                  style: Brand.captionFont.copyWith(
                    color: Brand.warmGray,
                    decoration: TextDecoration.underline,
                  ),
                ),
              ),
              Padding(
                padding: const EdgeInsets.symmetric(horizontal: 8),
                child: Text(
                  '|',
                  style: Brand.captionFont.copyWith(color: Brand.warmGray),
                ),
              ),
              GestureDetector(
                onTap: () {
                  // Open privacy policy
                },
                child: Text(
                  'Privacy Policy',
                  style: Brand.captionFont.copyWith(
                    color: Brand.warmGray,
                    decoration: TextDecoration.underline,
                  ),
                ),
              ),
            ],
          ),
        ],
      ),
    );
  }
}

// ---------------------------------------------------------------------------
// MARK: - Feature Row Model
// ---------------------------------------------------------------------------

class _FeatureRow {
  final String name;
  final String? freeValue;
  final String premiumValue;

  const _FeatureRow(this.name, this.freeValue, this.premiumValue);
}

File: Little Artist/Views/PaywallView.swift
//
//  PaywallView.swift
//  Little Artist
//
//  Shown when users hit free tier limits.
//  Integrates with StoreKitManager for real purchases.
//

import SwiftUI
import StoreKit

/// A modal paywall shown when a free tier limit is reached.
struct PaywallView: View {
    @Environment(\.dismiss) private var dismiss

    /// Describes which limit was hit.
    let reason: LimitReason

    private var store: StoreKitManager { StoreKitManager.shared }

    /// Dynamically calculated savings text based on live product prices.
    private var savingsText: String {
        guard let monthly = store.monthlyProduct,
              let yearly = store.yearlyProduct else {
            return "Best value"
        }
        let annualMonthly = NSDecimalNumber(decimal: monthly.price * 12).doubleValue
        let yearlyPrice = NSDecimalNumber(decimal: yearly.price).doubleValue
        guard annualMonthly > 0 else { return "Best value" }
        let savings = ((annualMonthly - yearlyPrice) / annualMonthly * 100).rounded(.down)
        return "Best value — save \(Int(savings))%"
    }

    enum LimitReason: Identifiable {
        var id: Self { self }
        case children
        case artworks

        var title: String {
            "Unlock Premium Features"
        }

        var message: String {
            switch self {
            case .children:
                return "Add unlimited child profiles, cloud sync, AI captions, and more with Artling Premium."
            case .artworks:
                return "Save unlimited artworks, sync across devices, and unlock AI-powered captions with Artling Premium."
            }
        }

        var icon: String {
            "sparkles"
        }
    }

    /// The purchase plans offered on the paywall.
    enum Plan {
        case monthly
        case yearly
        case lifetime
    }

    /// The currently selected plan.
    @State private var selectedPlan: Plan = .yearly

    // MARK: - Body

    var body: some View {
        ScrollView {
            VStack(spacing: Brand.sectionSpacing) {
                // MARK: Header
                header

                // MARK: Hero
                heroSection

                // MARK: Benefits
                benefitCards

                // MARK: Subscription Tiers
                subscriptionTiers

                // MARK: CTA
                ctaButton

                // MARK: Loading / Error States
                statusSection

                // MARK: Legal
                legalSection

                // MARK: Restore
                restoreButton
            }
            .padding(.vertical, Brand.screenPadding)
        }
        .background(BrandAppBackground())
        .task(id: store.isPremium) {
            if store.isPremium {
                dismiss()
            }
        }
        .onChange(of: store.isPremium) { _, isPremium in
            if isPremium { dismiss() }
        }
    }

    // MARK: - Header

    private var header: some View {
        HStack {
            Button {
                dismiss()
            } label: {
                Image(systemName: "xmark.circle.fill")
                    .font(.system(size: 28))
                    .foregroundStyle(Brand.warmGray.opacity(0.6))
            }

            Spacer()

            HStack(spacing: 6) {
                Image(systemName: "sparkles")
                    .font(.system(size: 16, weight: .semibold))
                    .foregroundStyle(Brand.primary)

                Text("Little Artist Premium")
                    .font(Brand.headlineFont)
                    .foregroundStyle(Brand.primary)
            }

            Spacer()

            // Invisible spacer to balance close button
            Color.clear
                .frame(width: 28, height: 28)
        }
        .padding(.horizontal, Brand.screenPadding)
    }

    // MARK: - Hero Section

    private var heroSection: some View {
        VStack(spacing: 8) {
            Text("Unlock Your Full Studio")
                .font(Brand.title1Font)
                .foregroundStyle(Brand.charcoal)
                .multilineTextAlignment(.center)

            Text(reason.message)
                .font(Brand.bodyFont)
                .foregroundStyle(Brand.warmGray)
                .multilineTextAlignment(.center)
                .padding(.horizontal, Brand.screenPadding)
        }
    }

    // MARK: - Benefit Cards

    private var benefitCards: some View {
        VStack(spacing: 12) {
            benefitCard(
                icon: "person.3",
                tint: Brand.primary,
                title: "Unlimited Child Profiles",
                description: "Track every little artist in the family"
            )
            benefitCard(
                icon: "square.grid.2x2",
                tint: Brand.sky,
                title: "Unlimited Artworks",
                description: "Never run out of space for masterpieces"
            )
            benefitCard(
                icon: "cloud.fill",
                tint: Brand.lavender,
                title: "Cloud Backup & Sync",
                description: "Keep artwork safe across all your devices"
            )
            benefitCard(
                icon: "doc.text",
                tint: Brand.warmGray,
                title: "PDF Portfolio Export",
                description: "Create beautiful keepsake portfolios"
            )
            // Emphasized card for Voice Memo & AI
            benefitCard(
                icon: "mic.fill",
                tint: Brand.primary,
                title: "Voice Memo & AI",
                description: "Record stories and get AI-powered captions",
                emphasized: true
            )
        }
        .padding(.horizontal, Brand.screenPadding)
    }

    // MARK: - Subscription Tiers

    private var subscriptionTiers: some View {
        VStack(spacing: 12) {
            // Yearly tier — highlighted with savings badge
            Button {
                selectedPlan = .yearly
                HapticService.selection()
            } label: {
                VStack(spacing: 0) {
                    // Best Value badge
                    Text(savingsText)
                        .font(Brand.caption2Font)
                        .fontWeight(.semibold)
                        .foregroundStyle(.white)
                        .padding(.horizontal, 14)
                        .padding(.vertical, 5)
                        .background(Brand.lavender, in: Capsule())
                        .offset(y: -2)

                    VStack(spacing: 4) {
                        Text("Yearly")
                            .font(Brand.headlineFont)
                            .foregroundStyle(Brand.charcoal)

                        Text(store.yearlyProduct?.displayPrice ?? "$34.99")
                            .font(Brand.title2Font)
                            .foregroundStyle(Brand.primary)

                        Text("per year")
                            .font(Brand.caption2Font)
                            .foregroundStyle(Brand.warmGray)
                    }
                    .padding(.vertical, Brand.buttonPadding)
                    .frame(maxWidth: .infinity)
                }
                .background(Brand.surface)
                .clipShape(RoundedRectangle(cornerRadius: Brand.radiusCard, style: .continuous))
                .overlay(
                    RoundedRectangle(cornerRadius: Brand.radiusCard, style: .continuous)
                        .strokeBorder(
                            selectedPlan == .yearly ? Brand.lavender : Brand.softTan,
                            lineWidth: selectedPlan == .yearly ? 2.5 : 1
                        )
                )
                .brandCardShadow()
            }
            .buttonStyle(.plain)

            // Monthly and Lifetime — side by side
            HStack(spacing: 12) {
                compactTierCard(
                    plan: .monthly,
                    name: "Monthly",
                    price: store.monthlyProduct?.displayPrice ?? "$4.99",
                    detail: "per month"
                )

                compactTierCard(
                    plan: .lifetime,
                    name: "Lifetime",
                    price: store.lifetimeProduct?.displayPrice ?? "$79.99",
                    detail: "pay once, forever"
                )
            }
        }
        .padding(.horizontal, Brand.screenPadding)
    }

    /// A compact selectable tier card for the monthly and lifetime plans.
    private func compactTierCard(
        plan: Plan,
        name: String,
        price: String,
        detail: String
    ) -> some View {
        Button {
            selectedPlan = plan
            HapticService.selection()
        } label: {
            VStack(spacing: 4) {
                Text(name)
                    .font(Brand.headlineFont)
                    .foregroundStyle(Brand.charcoal)

                Text(price)
                    .font(Brand.title2Font)
                    .foregroundStyle(selectedPlan == plan ? Brand.primary : Brand.charcoal)

                Text(detail)
                    .font(Brand.caption2Font)
                    .foregroundStyle(Brand.warmGray)
            }
            .padding(.vertical, Brand.buttonPadding)
            .frame(maxWidth: .infinity)
            .background(Brand.surface)
            .clipShape(RoundedRectangle(cornerRadius: Brand.radiusCard, style: .continuous))
            .overlay(
                RoundedRectangle(cornerRadius: Brand.radiusCard, style: .continuous)
                    .strokeBorder(
                        selectedPlan == plan ? Brand.primary : Brand.softTan,
                        lineWidth: selectedPlan == plan ? 2.5 : 1
                    )
            )
            .brandCardShadow()
        }
        .buttonStyle(.plain)
    }

    // MARK: - CTA Button

    /// The product matching the currently selected plan, if loaded.
    private var selectedProduct: Product? {
        switch selectedPlan {
        case .monthly: return store.monthlyProduct
        case .yearly: return store.yearlyProduct
        case .lifetime: return store.lifetimeProduct
        }
    }

    private var ctaButton: some View {
        VStack(spacing: 10) {
            Button {
                Task {
                    if let product = selectedProduct {
                        await store.purchase(product)
                    }
                }
            } label: {
                Text(selectedPlan == .lifetime ? "Unlock Forever" : "Subscribe Now")
                    .font(Brand.headlineFont)
                    .foregroundStyle(.white)
                    .frame(maxWidth: .infinity)
                    .padding(.vertical, Brand.buttonPadding)
                    .background(Brand.primary)
                    .clipShape(Capsule())
                    .brandFABShadow()
            }
            .disabled(store.isPurchasing || store.products.isEmpty)

            Text("Cancel anytime · No hidden fees")
                .font(Brand.caption2Font)
                .foregroundStyle(Brand.warmGray)
        }
        .padding(.horizontal, Brand.screenPadding)
    }

    // MARK: - Status Section

    @ViewBuilder
    private var statusSection: some View {
        if store.products.isEmpty {
            ProgressView()
                .tint(Brand.primary)
                .padding(.vertical, 8)
        }

        if store.isPurchasing {
            ProgressView("Processing...")
                .font(Brand.captionFont)
                .tint(Brand.primary)
        }

        if let error = store.errorMessage {
            Text(error)
                .font(Brand.captionFont)
                .foregroundStyle(Brand.dustyRose)
                .multilineTextAlignment(.center)
                .padding(.horizontal, Brand.formPadding)
        }
    }

    // MARK: - Legal Section

    private var legalSection: some View {
        VStack(spacing: 6) {
            Text("Subscriptions automatically renew unless cancelled at least 24 hours before the end of the current period. Manage subscriptions in your Apple Account subscriptions settings. Lifetime is a one-time purchase and never renews.")
                .font(Brand.caption2Font)
                .foregroundStyle(Brand.warmGray)
                .multilineTextAlignment(.center)

            HStack(spacing: 16) {
                Link("Privacy Policy", destination: URL(string: "https://www.flutterly.co.uk/projects/artling/privacy-policy")!)
                Link("Terms of Use", destination: URL(string: "https://www.apple.com/legal/internet-services/itunes/dev/stdeula/")!)
            }
            .font(Brand.caption2Font)
            .foregroundStyle(Brand.warmGray)
        }
        .padding(.horizontal, Brand.formPadding)
    }

    // MARK: - Restore Button

    private var restoreButton: some View {
        Button {
            Task { await store.restorePurchases() }
        } label: {
            Text("Restore Purchases")
                .font(Brand.captionFont)
                .foregroundStyle(Brand.warmGray)
        }
        .padding(.bottom, 8)
    }

    // MARK: - Benefit Card Helper

    private func benefitCard(
        icon: String,
        tint: Color,
        title: String,
        description: String,
        emphasized: Bool = false
    ) -> some View {
        HStack(spacing: 14) {
            // Icon in tinted circle
            Image(systemName: icon)
                .font(.system(size: emphasized ? 18 : 16, weight: .semibold))
                .foregroundStyle(tint)
                .frame(width: emphasized ? 44 : 40, height: emphasized ? 44 : 40)
                .background(tint.opacity(0.12), in: Circle())

            VStack(alignment: .leading, spacing: 2) {
                Text(title)
                    .font(emphasized ? Brand.headlineFont : Brand.subheadlineFont.weight(.semibold))
                    .foregroundStyle(Brand.charcoal)

                Text(description)
                    .font(Brand.captionFont)
                    .foregroundStyle(Brand.warmGray)
            }

            Spacer()
        }
        .padding(emphasized ? 16 : 14)
        .background(Brand.surface)
        .clipShape(RoundedRectangle(cornerRadius: Brand.radiusCard, style: .continuous))
        .overlay(
            RoundedRectangle(cornerRadius: Brand.radiusCard, style: .continuous)
                .strokeBorder(emphasized ? Brand.primary.opacity(0.3) : Brand.glassStroke, lineWidth: 1)
        )
        .brandCardShadow()
    }
}

// MARK: - Preview

#Preview("Children Limit") {
    PaywallView(reason: .children)
}

#Preview("Artwork Limit") {
    PaywallView(reason: .artworks)
}
